python/node_classification.py

```diff
@@ -0,0 +1,342 @@
+from mage.node_classification.utils.convert_data import convert_data
+import torch
+import numpy as np
+import sys
+import mgp
+
+##############################
+# constants
+##############################
+
+
+class ModelParams:
+    IN_CHANNELS = "in_channels"
+    OUT_CHANNELS = "out_channels"
+    HIDDEN_FEATURES_SIZE = "hidden_features_size"
+    LAYER_TYPE = "layer_type"
+    AGGREGATOR = "aggregator"
+
+
+class OptimizerParams:
+    OPTIMIZER = "optimizer"
+    LEARNING_RATE = "learning_rate"
+    WEIGHT_DECAY = "weight_decay"
+
+
+class DataParams:
+    SPLIT_RATIO = "split_ratio"
+    METRICS = "metrics"
+
+
+class MemgraphParams:
+    NODE_FEATURES_PROPERTY = "node_features_property"
+    NODE_ID_PROPERTY = "node_id_property"
+    NODE_CLASS_PROPERTY = "node_class_property"
+
+
+class TrainParams:
+    NUM_EPOCHS = "num_epochs"
+    CONSOLE_LOG_FREQ = "console_log_freq"
+    CHECKPOINT_FREQ = "checkpoint_freq"
+    TOTAL_NO_EPOCHS = "total_no_epochs"
+
+
+class OtherParams:
+    DEVICE_TYPE = "device_type"
+    PATH_TO_MODELS = "/home/mateo/mage/pytorch_models/model"
+
+
+# all None until set_params are executed
+data: Data = None
+model: InductiveModel = None
+opt = None
+criterion = None
+global_params: mgp.Map
+logged_data: mgp.List = []
+
+DEFINED_INPUT_TYPES = {
+    ModelParams.HIDDEN_FEATURES_SIZE: list,
+    ModelParams.LAYER_TYPE: str,
+    TrainParams.NUM_EPOCHS: int,
+    OptimizerParams.OPTIMIZER: str,
+    OptimizerParams.LEARNING_RATE: float,
+    OptimizerParams.WEIGHT_DECAY: float,
+    DataParams.SPLIT_RATIO: float,
+    MemgraphParams.NODE_FEATURES_PROPERTY: str,
+    MemgraphParams.NODE_ID_PROPERTY: str,
+    OtherParams.DEVICE_TYPE: str,
+    TrainParams.CONSOLE_LOG_FREQ: int,
+    TrainParams.CHECKPOINT_FREQ: int,
+    ModelParams.AGGREGATOR: str,
+    DataParams.METRICS: list
+}
+
+DEFAULT_VALUES = {
+    ModelParams.HIDDEN_FEATURES_SIZE: [16],
+    ModelParams.LAYER_TYPE: "SAGE",
+    TrainParams.NUM_EPOCHS: 100,
+    OptimizerParams.OPTIMIZER: "Adam",
+    OptimizerParams.LEARNING_RATE: 0.1,
+    OptimizerParams.WEIGHT_DECAY: 5e-4,
+    DataParams.SPLIT_RATIO: 0.8,
+    MemgraphParams.NODE_FEATURES_PROPERTY: "features",
+    MemgraphParams.NODE_ID_PROPERTY: "id",
+    MemgraphParams.NODE_CLASS_PROPERTY: "class",
+    OtherParams.DEVICE_TYPE: "cpu",
+    TrainParams.CONSOLE_LOG_FREQ: 5,
+    TrainParams.CHECKPOINT_FREQ: 5,
+    ModelParams.AGGREGATOR: "mean",
+    DataParams.METRICS: ["loss", "accuracy", "f1_score", "precision", "recall", "num_wrong_examples"],
+}
+
+
+##############################
+# set model parameters
+##############################
+
+def declare_globals(params: mgp.Map):
+    """This function declares dictionary of global parameters to given dictionary.
+
+    Args:
+        params (mgp.Map): given dictionary of parameters
+    """
+    global global_params
+    global_params = params
+
+
+def declare_model_and_data(ctx: mgp.ProcCtx):
```

Comment:
The only thing that should be global is the model. This method should return all relevant information but maybe wrapped into an object, and not communicated via global vars.

Reply:
I wrapped all those global variables in one object ´Modelling´.